expressionGenerator.js

```diff
@@ -0,0 +1,51 @@
+const fieldsToUpdate = (body) => {
+    var fields = ['name', 'description', 'showDate']
+    var fieldsToUpdate = []
+    var expressions = []
+    const bodyFields = Object.keys(body)
+
+    for (let index = 0; index < bodyFields.length; index++) {
+        if (fields.includes(bodyFields[index])) {
+            fieldsToUpdate.push(bodyFields[index])
+        }
+    }
```

Comment:
Massa !! Curto usar o `Object.keys(...)` também

O que indico ?? isso aqui faz uma coisa só e pode ser uma função separada. Fora da função principal. algo do tipo

No template, passa o seu ['name', 'description'...] e no fields passa o seu `Object.keys(...)`
`const getFieldsByTemplate = (template, fields) => {....}`

Mas falando da lógica aqui, o JS tem umas funções mais legais pra usar, mais rápidas também segue um exemplo do filter:
```suggestion
    const bodyFields = Object.keys(body)
    const fieldsToUpdate = bodyFields.filter((field) => fields.includes(field))
```